airflow/utils/timezone.py

```diff
@@ -184,3 +185,38 @@ def coerce_datetime(v: Union[None, dt.datetime, DateTime]) -> Optional[DateTime]
     if v.tzinfo is None:
         v = make_aware(v)
     return pendulum.instance(v)
+
+
+def _get_tzname_or_offset(obj):
+    if _is_valid_pendulum_tzname(obj):
+        return obj
+    if hasattr(obj, 'name'):
+        candidate = obj.name
+        if _is_valid_pendulum_tzname(candidate):
+            return candidate
+    try:
+        candidate = obj.tzname(dt.datetime.now(dt.timezone.utc))
+        if _is_valid_pendulum_tzname(candidate):
+            return candidate
+    except (NotImplementedError, ValueError):
+        pass
+
+    try:
+        candidate = int(obj.utcoffset(dt.datetime.now(dt.timezone.utc)).total_seconds())
```

Comment:
@ashb do we want to do this or not? at this point there is no real guarantee that the `datetime.tzinfo` that is really a fixed offset timezone. There is no good way to check if it really is. I can do an `isinstance(obj, pendulum.tz.timezone.FixedTimezone)` that works only for pendulum timezones.

 The risk is that someone uses some custom `datetime.tzinfo` with a custom name with no equivalent in pendulum, and this serialization logic will reduce that custom tzinfo to a fixedoffset timezone (which will have a different behaviour than the original dag.timezone).

Reply:
> The risk is that someone uses some custom datetime.tzinfo with a custom name with no equivalent in pendulum

I think so long as we document "don't do this"  I'm okay wit this. The chance of someone creating a custom timezone seems _very_ slim given Airflow's typical users.

Reply:
https://airflow.apache.org/docs/apache-airflow/stable/timezone.html?highlight=timezone#time-zone-aware-dags would be the place to put such a comment.

Reply:
Ok, I added a "Don't use other `datetime.tzinfo` implementations" caveat on that page (timezone.rst)